src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -110,6 +112,7 @@ export class ReadingsGraphComponent implements OnDestroy {
       activeDropDowns[0].classList.remove('is-active');
     }
     this.optedTime = ASSET_READINGS_TIME_FILTER;
+    super.ngOnDestroy(); // call to unsubscribes all subscriptions
```

Comment:
I'm slightly worried, this is calling ngOnDestroy of component. While we are just toggling the modal (part of component). May be there are only modal level variables only today but that is component level life-cycle event.

Am I missing anything?